src/GAP/gap_to_oscar.jl

```diff
@@ -43,3 +43,29 @@ function GAP.gap_to_julia(::Type{fmpq_mat}, obj::GAP.GapObj)
 end
 
 fmpq_mat(obj::GAP.GAP.GapObj) = gap_to_julia(fmpq_mat, obj)
+
+## nonempty list of GAP matrices over the same cyclotomic field
+function matrices_over_cyclotomic_field(gapmats::GAP.GapObj)
+    GAP.Globals.IsList(gapmats) || throw(ArgumentError("gapmats is not a GAP list"))
+    GAP.Globals.IsEmpty(gapmats) && throw(ArgumentError("gapmats is empty"))
+    GAP.Globals.ForAll(gapmats, GAP.Globals.IsCyclotomicCollColl) ||
+      throw(ArgumentError("gapmats is not a GAP list of matrices of cyclotomics"))
+
+    gapF = GAP.Globals.FieldOfMatrixList(gapmats)
+    N = GAP.Globals.Conductor(gapF)
+    root = GAP.Globals.E(N)
+    elms = [root^i for i in 0:(euler_phi(N)-1)]
+    B = GAP.Globals.Basis(gapF, GAP.GapObj(elms))
+    F, z = CyclotomicField(N)
```

Comment:
How about also adding a method which enables one to write `F(x)` where F is a cyclotomic field and `x` a GAP cyclotomic? I.e., something like this:
```julia
function (a::AnticNumberField)(x::GapObj)
   GAP.Globals.IsCyclotomic(b) || throw(ArgumentError("input is not a GAP cyclotomic"))
   B = [... compute basis like above... perhaps we could also cache these in a `WeakValueDict` from AbstractAlgebra ]
  return F(Vector{fmpz}(GAP.Globals.Coefficients(B, x)))
end
```

Granted, we probably wouldn't want to use it in `matrices_over_cyclotomic_field` for performance reasons (recomputing the basis again and again is bad -- unless we cache it...), but it might still be useful for some users, no?

Hmm, and I guess taking this one step further, perhaps a `matrix(F, mat)` (where `mat` is the GAP matrix) could/should be made to work?